app/src/main/java/io/appium/uiautomator2/model/NotificationListener.java

```diff
@@ -67,6 +67,7 @@ public void stop() {
             return;
         }
         Logger.debug("Stopping toast notification listener.");
+        isListening = false;
         uiAutomation.setOnAccessibilityEventListener(originalListener);
```

Comment:
is it ok to set the listener to null even if we cannot receive the original value?

Reply:
We have no other way to remove our listener. ```Null``` is ok.
https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/android/app/UiAutomation.java#1139